### PR TITLE
feat: use ory themes for ui

### PR DESCRIPTION
**Pull request by Benehiko**
This PR overhauls the UI

The classes are generated from `@ory/themes` and then injected into the HTML templates using vanilla JS. I did not "insall" the library from npm or git, but rather copied what i needed into this project since we are short on time.

Registration Page
image

Login Page
image

Error Page
image

Index Page
image

image
image

### CODE REVIEW

File: backend/handler.go [hunk 1]
```diff
@@ -64,6 +65,7 @@ func NewHandler(repo *sqlRepo, config *Config) (http.Handler, error) {
 	r.POST("/grant-access", h.grantAccess)
 	r.GET("/flag", h.getFlag)
 	r.POST("/flag", h.submitFlag)
+	r.GET("/static/*filepath", h.static)
```
Comment (Benehiko) [inline backend/handler.go:68]: Here I register the files using an embed filesystem.
So all the files we serve to the end-user will be served from within the compiled binary. This should help mitigate deployment problems

File: backend/handler_flag.go [hunk 2]
```diff
@@ -4,18 +4,24 @@ import (
...
 	"strings"
 	"time"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/ory/herodot"
+	rts "github.com/ory/keto/proto/ory/keto/relation_tuples/v1alpha2"
 )
 
+//go:embed static/*
+var staticFiles embed.FS
```
Comment (Benehiko) [inline backend/handler_flag.go:23]: loading directory into embed

File: configs/oathkeeper_rules.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
       "url": "http://backend:8000"
     },
     "match": {
-      "url": "http://localhost:5050/<login|register|error>",
+      "url": "http://localhost:5050/<login|register|error|(static/.*)>",
```
Comment (Benehiko) [inline configs/oathkeeper_rules.json:8]: map static route in oathkeeper rules

File: backend/static/theming.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+const buttonLinks = document.querySelectorAll(".button-link");
+for (const e of buttonLinks) {
+  e.classList.add(...buttonLinkStyle().split(" "));
+}
+
+const errorMessages = document.querySelectorAll(".error-message");
+for (const e of errorMessages) {
+  e.classList.add(...messageStyle({ severity: "error" }).split(" "));
+}
```
Comment (Benehiko) [inline backend/static/theming.js:88]: This file does the theme injection. 
It basically gets the generated class names from our bundled index.es.js file which vite bundles with vanilla-extract. the names are generated, so this is required for us to map the correct classes to the correct html elements.

File: backend/handler_auth.go [hunk 5]
```diff
@@ -116,7 +117,7 @@ func getFlow[F interface {
 	}
 	data.Messages = string(msgRaw)
 	if len(msg) == 0 {
-		data.Messages = "no messages right now"
+		data.Messages = "no error messages right now"
```
Comment (Benehiko) [inline backend/handler_auth.go:120]: Not sure if we want to display this text all the time? but we can keep it i guess :)

**Review by zepatrik — APPROVED** [review]

**Pull request merged**